app/services/sendgrid_api.rb

```diff
@@ -0,0 +1,39 @@
+class SendgridApi
+  extend SingleForwardable
+
+  def_single_delegators :new, :spam_reported?, :bounced?,
+    :remove_from_bounce_list, :remove_from_spam_list
+
+  def spam_reported?(email)
+    api { spam_reports.retrieve(email: email).any? }
+  end
+
+  def bounced?(email)
+    api { bounces.retrieve(email: email).any? }
+  end
+
+  def remove_from_bounce_list(email)
+    api { bounces.delete(email: email) }
+  end
+
+  def remove_from_spam_list(email)
+    api { spam_reports.delete(email: email) }
+  end
+
+private
+
+  def api(&_action)
+    yield
+  rescue => e
+    Rails.logger.error("SendgridApi error: #{e.class} #{e}")
+    false
+  end
+
+  def spam_reports
+    SendgridToolkit::SpamReports.new
```

Comment:
It seems a bit magical having the toolkit perform the environment variable lookup for the username & password - I was only aware of this feature after looking at the project.

I'd sooner it be set in the project - we could have a sendgrid initializer that grabs the ENV vars and sets this up for us.